src/antlr/index.ts

```diff
@@ -10,7 +10,7 @@ import { CompletionVisitor } from "./CompletionVisitor";
 import { Cursor } from "./Cursor";
 import _ from "lodash";
 
-interface ParserOptions {
+export interface ParserOptions {
   doubleQuotedIdentifier?: boolean;
   cursorPosition?: { lineNumber: number; column: number };
```

Comment:
@megaflo Would help to add a comment here. Why provide a cursor position?

Reply:
The position is needed due to the way we handle the cursor. To make sure the parser doesn't fail in common "completion scenarios" (such as a trailing comma in the `SELECT` clause) we add a placeholder that the parser can deal with. This placeholder will also be used to then identify the actual completion context in the AST. So we need the position to insert the placeholder.

I'll add a comment.

Reply:
@megaflo We use the same `parse` command for lineage and for auto-completion, however, we only care about the cursor in auto-completion right?

Reply:
Yes. We would need a separate parse method, if we don't want to allow it for lineage. The one reason I can think of, to leave it as it is, would be that right now we can compute completions and lineage from the same parsed query. In theory we could use this in the ui to populate two components at the same time.

Reply:
Hmm... we could do a bit of inheritance here to clean up the api:
```typescript
class SqlParseTree {
  getUsedTables() { ... }
  getLineage(...) { ... }
}

class CompletableSqlParseTree extends SqlParseTree {
  getSuggestions(...) { ... }
}

const antlr = {
  parseForLineage(...): SqlParseTree { ... }
  parseForCompletion(...): CompletableSqlParseTree { ... }
};
```
WDYT?

Reply:
@megaflo looks good to me 👍